src/main/java/icbm/classic/content/blast/BlastGasBase.java

```diff
@@ -307,10 +307,27 @@ private void setEffectBoundsAndSpawnParticles(int timePassed)
         edgeBlocks.addAll(nextSet);
     }
 
-    private boolean isValidPath(BlockPos pos)
+    private boolean isValidPath(BlockPos pos, EnumFacing direction)
     {
         IBlockState blockState = world.getBlockState(pos);
-        return !blockState.isFullBlock();
+        final AxisAlignedBB aabb = blockState.getCollisionBoundingBox(world, pos);
+        if (aabb == null) // if there is no bounding box, its pass through, so its a valid path
+            return true;
+
+        // whether or not the respective axes are completely stretched (they span form one side of the block to another)
+        boolean xFull = aabb.minX==0 && aabb.maxX == 1;
+        boolean yFull = aabb.minY==0 && aabb.maxY == 1;
+        boolean zFull = aabb.minZ==0 && aabb.maxZ == 1;
+
+        boolean isImpassable = false;
+        if (direction == EnumFacing.UP || direction == EnumFacing.DOWN)
+            isImpassable = xFull && zFull;
```

Comment:
I think there is a helper for this to check if a face is solid. Look at the torch block and see if you can find it. Though in the future we want to expand this to decide which blocks are air tight.